### PR TITLE
Added a ScriptAttribute class

### PR DESCRIPTION
**Pull request by zorg93**
Needs work, but a basic ScriptAttribute that will produce Log Information.
Displays the Version number in Log file that will help a developer know if a user experienced a bug/error on the latest or an outdated version.
Displaying a support url when a script crashes/fails to start to point the user in the direction to get help

### CODE REVIEW

**Comment by JohnnyCrazy** [comment]
Mh, I'm not a big fan of specifing the version with scripts.  
DLLs should have the version in the assembly, .cs/.vb files should have them in the name.

Also, versions should be script independent, shouldnt they?

**Comment by zorg93** [comment]
I thought about assembley but the vb/cs wouldnt support that. The idea was something that would be the same across both ways of distributing a mod

**Comment by crosire** [comment]
I agree with @JohnnyCrazy there. Loose script files should mainly be used for prototyping, anything else is expected to be a DLL really. The support string is a nice addition. The whole attribute idea might get interesting for the console later too for additional information.

**Comment by zorg93** [comment]
OK I see where you're coming from. This should be a much nicer setup. When the scriptdomain loads the assembley, if its a dll, then it will also print the version extracted from the dll next to its file name. This has the added advantage that libraries referenced also show version info, so if someone is using an outdated version of the library. the developer would be able to see thats what is causing the issue.

 Of course this will only work if people actually start putting version information in their manifest files

**Comment by zorg93** [comment]
quick example of how this looks
[13:18:51] [INFO] Found 0 script(s) in 'test.cs'.
[13:18:51] [INFO] Found 1 script(s) in 'NativeUI.dll v1.0.0'.

File: source/core/Script.hpp [hunk 1]
```diff
@@ -33,6 +33,29 @@ namespace GTA
 		System::Type ^_dependency;
 	};
 
+	public ref class StriptAttribute : System::Attribute
```
Comment (crosire) [inline source/core/Script.hpp]: This should be `ScriptAttribute`, right? :smile:

**Comment by crosire** [comment]
I wonder if it wouldn't be easier to provide some overridable properties in the `Script` class instead.

**Comment by zorg93** [comment]
@crosire Doing it that way would mean the script would need to be instantiated before you could grab the properties, which would be an issue if the constructor threw an error

File: source/core/Script.hpp (hunk 1 above)
Comment (zorg93) [inline source/core/Script.hpp]: I swear I'm not dyslexic

**Comment by crosire** [comment]
@zorg93 Good point.

**Pull request merged**